### ISSUE TITLE
Fatal: memory_linux.cc(35) Out of Memory

### ISSUE THREAD

**Issue by ratsplayer** [philipptrenz/photo-booth#98]
Hi,

at first: You created a very nice project.

I am using a Raspberry 3B+ and installed the photo-booth as shown in the Readme.
It starts and seems to work, i also can see photos and create a new one.
But at taking the second photo it only shows a black screen and stops.
In the console i can see the following error text:

FATAL:memory_linux.cc(35)] Out of memory.
#0 0x0000... <unknown>
#1 ...
#2 
---
#4 0x... calloc

I think it runs out of memory.
If have to say i am using the simulation mode from marcel hofer.
At the moment i don't have a compatible camera.

Do you have any idea how to increase memory or fix that problem?

Thank you very much!

**Comment by marcel-hofer** [comment]
Thanks for reporting the issue.

I created this simulation feature mainly for development on my windows pc and apparently didn't test it so well on my PI.

I'm also using a Raspberry PI 3B+ and was able to reproduce the out of memory exception you reported.
For me it always failed during the second sample image generation.

I intentionally created large sample images (6000x4000) to simulate high camera resolutions.
They result in about 200KB files but that should not be a problem compared to real pictures sizes.
After a few tests it looks like the combination of *creating a red picture* and *set timestamp watermark using svg* (composite) with the high resolution is the problem.

Therefore i lowered the resolution (results in faster processing time on the Raspberry PI) and merged the operations together which should fix the exception.
After successfully taking 20+ sample images I think it works as needed (you probably won't create that much pictures in simulation mode).

I fixed the issue in my repo (develop + printing-gif-translations branches) and updated the open pull request #95
